timeutil/human.go

```diff
@@ -0,0 +1,56 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package timeutil
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/snapcore/snapd/i18n"
+)
+
+func noon(t time.Time) time.Time {
+	y, m, d := t.Date()
+	return time.Date(y, m, d, 12, 0, 0, 0, t.Location())
+}
+
+// Human turns the time into a relative expression of time meant for human
+// consumption.
+// Human(t)  --> "today at 07:47 "
+func Human(then time.Time) string {
+	return humanTimeSince(then.Local(), time.Now().Local())
+}
+
+func humanTimeSince(then, now time.Time) string {
+	d := int(math.Floor(noon(then).Sub(noon(now)).Hours() / 24))
+	switch {
+	case d > 1:
```

Comment:
Can you please order this so that the messages are first in the past and then in the future. Now it's mixed and that's harder to read IMO.

Reply:
sure. I ordered it like this because then the more common cases don't have to walk through the less common ones :-) but it's a silly thing to do if it impacts readability